source/Octopus.Tentacle/Maintenance/WorkspaceCleaner.cs

```diff
@@ -51,9 +48,10 @@ public async Task Clean(CancellationToken cancellationToken)
 
                 try
                 {
-                    if (scriptService.IsRunningScript(workspace.ScriptTicket)) continue;
-                    if (scriptServiceV2.IsRunningScript(workspace.ScriptTicket)) continue;
-                    if(scriptServiceV3Alpha.IsRunningScript(workspace.ScriptTicket)) continue;
+                    if (runningScriptReporters.Any(x => x.IsRunningScript(workspace.ScriptTicket)))
```

Comment:
Interesting - each service keeps its own running scripts in memory.

This is one more in-memory thing we need to take out for K8s?

Reply:
The implementation in the k8s script service will just look for an existing pod via the PodStatusMonitor